src/v3/src/transformer/layout/oktaVerify/transformOktaVerifyDeviceChallengePoll.ts

```diff
@@ -12,60 +12,158 @@
 } from '../../../types';
 import { loc } from '../../../util';
 
+const getTitleText = (challengeMethod = CHALLENGE_METHOD.CUSTOM_URI) => {
+  if (challengeMethod === CHALLENGE_METHOD.APP_LINK) {
+    return loc('appLink.title', 'login');
+  } if (challengeMethod === CHALLENGE_METHOD.UNIVERSAL_LINK) {
+    return loc('universalLink.title', 'login');
+  }
+  return loc('customUri.title', 'login');
+};
+
+const getDescriptionText = (challengeMethod = CHALLENGE_METHOD.CUSTOM_URI) => {
+  if (challengeMethod === CHALLENGE_METHOD.APP_LINK) {
+    return loc('appLink.content', 'login');
+  } if (challengeMethod === CHALLENGE_METHOD.UNIVERSAL_LINK) {
+    return loc('universalLink.content', 'login');
+  }
+  return loc('customUri.required.content.prompt', 'login');
+};
+
 export const transformOktaVerifyDeviceChallengePoll: IdxStepTransformer = ({
   transaction,
   formBag,
+  prevTransaction,
 }) => {
   const { nextStep = {} as NextStep } = transaction;
   const { uischema } = formBag;
+  const FASTPASS_FALLBACK_SPINNER_TIMEOUT = 4000;
 
   const deviceChallengePayload = transaction.nextStep?.name === IDX_STEP.DEVICE_CHALLENGE_POLL
     ? transaction.nextStep?.relatesTo?.value
     // @ts-expect-error challenge is not defined on contextualData
     : transaction.nextStep?.relatesTo?.value?.contextualData?.challenge?.value;
 
-  uischema.elements.unshift({
+  const { challengeMethod } = deviceChallengePayload;
+  const titleElement: TitleElement = {
     type: 'Title',
-    options: { content: loc('customUri.title', 'login') },
-  } as TitleElement);
+    options: {
+      content: getTitleText(challengeMethod),
+    },
+  };
 
-  uischema.elements.push({
+  const descriptionElement: DescriptionElement = {
     type: 'Description',
     contentType: 'subtitle',
-    options: { content: loc('customUri.required.content.prompt', 'login') },
-  } as DescriptionElement);
+    options: {
+      content: getDescriptionText(challengeMethod),
+    },
+  };
 
-  uischema.elements.push({
+  const spinnerElement = {
+    type: 'Spinner',
+  } as SpinnerElement;
+
+  const openOktaVerifyButton: OpenOktaVerifyFPButtonElement = {
     type: 'OpenOktaVerifyFPButton',
     options: {
       step: nextStep.name,
       href: deviceChallengePayload.href,
+      challengeMethod,
     },
-  } as OpenOktaVerifyFPButtonElement);
+  };
 
-  uischema.elements.push({
-    type: 'Description',
-    contentType: 'subtitle',
-    options: { content: loc('customUri.required.content.download.title', 'login') },
-  } as DescriptionElement);
-
-  uischema.elements.push({
+  const cancelStep = transaction.nextStep?.name === IDX_STEP.DEVICE_CHALLENGE_POLL
+    ? 'authenticatorChallenge-cancel' : 'currentAuthenticator-cancel';
+  const cancelLink: LinkElement = {
     type: 'Link',
+    contentType: 'footer',
     options: {
-      label: loc('customUri.required.content.download.linkText', 'login'),
-      href: deviceChallengePayload.downloadHref,
+      label: loc('goback', 'login'),
```

Comment:
I think updating the key used here will fix the issue with the test, mimicking what is used in the v2 view [here](https://github.com/okta/okta-signin-widget/blob/master/src/v2/view-builder/views/device/DeviceChallengePollView.js#L61)

Reply:
I was able to resolve the test issue by updating the key in the FPLoopback transformer aswell as in the device challenge transformer